components/templates/HomeTemplate.tsx

```diff
@@ -1,9 +1,38 @@
-import { Button, TextField } from '@mui/material';
+import { AddBoxOutlined, ExitToAppOutlined, LocalOfferOutlined, MailOutlined } from '@mui/icons-material';
+import { Button, Divider, TextField, Typography } from '@mui/material';
 import Head from 'next/head';
+import React from 'react';
 import 'tailwindcss/tailwind.css';
+import { DrawerMenu } from '../../components/compounds/DrawerMenu';
+import { DrawerMenuItem } from '../../components/molecules/DrawerMenuItem';
 import { TopLevelTemplate } from '../../@types';
 
 export const HomeTemplate: TopLevelTemplate = (props) => {
+  const menu = () => {
```

Comment:
`Consider: const Menu  = (props) => (<>{...stuff here}</>)`

So that

```
<>
  <Menu/>
</>
```

note: Does this still rerender?
Or!

useCallback / useMemo

Reply:
Consider extracting header